mrobosub_localization/src/localization.py

```diff
@@ -67,7 +72,8 @@ def handle_reset(self):
     def raw_depth_callback(self, raw_depth: Float32):
         if self.heave_offset is None:
             self.heave_offset = raw_depth.data
-        self.heave_pub.publish(raw_depth.data - self.heave_offset)
+        self.iekf_class.add_depth_measurement(raw_depth.data)
+        self.publish_state() 
```

Comment:
I'm not sure if we want to publish_state for every depth callback, imu callback and dvl callback it might just be better to have run publish_state periodically hopefully with the new data for all fields but this might have to be done combined, I'll add Arnav to review and he can double check

Reply:
I asked to put it on every callback, I don't really see a reason not to publish as soon as you have a new state estimate.

At most it'll be outputting at like 30-40 hz, which is still quite reasonable imo